commerce_coordinator/apps/orders/tests/test_views.py

```diff
@@ -0,0 +1,114 @@
+    client_class = APIClient
+
+    # Define test user properties
+    test_user_username = 'test'
+    test_user_email = 'test@example.com'
+    test_user_password = 'secret'
+
+    def setUp(self):
+        """Create test user before test starts."""
+
+        super().setUp()
+        User = get_user_model()
+        User.objects.create_user(self.test_user_username, self.test_user_email, self.test_user_password)
+
+    def tearDown(self):
+        """Log out any user from client after test ends."""
+
+        super().tearDown()
+        self.client.logout()
+
+    def test_ecommerce_view_rejects_post(self, mock_ecommerce_client):
+        """Check POST from authorized user receives a 405 Method Not Allowed."""
+
+        # Login
+        self.client.login(username=self.test_user_username, password=self.test_user_password)
+
+        # Perform POST
+        response = self.client.post(reverse('orders:order_history'), ECOMMERCE_REQUEST_GET_PARAMETERS)
+
+        # Check 405 Method Not Allowed
+        self.assertEqual(response.status_code, 405)
+
+    def test_ecommerce_view_rejects_unauthorized(self, mock_ecommerce_client):
+        """Check unauthorized users querying orders are redirected to login page."""
+
+        # Perform GET without logging in.
+        response = self.client.get(reverse('orders:order_history'), ECOMMERCE_REQUEST_GET_PARAMETERS)
+
+        # Check 302 Found with redirect to login page.
+        self.assertEqual(response.status_code, 302)
+        self.assertIn("/login", response.url)
+
+    def test_ecommerce_view_returns_ok(self, mock_ecommerce_client):
+        """Check authorized user querying orders receives an HTTP 200 OK."""
+
+        # Login
+        self.client.login(username=self.test_user_username, password=self.test_user_password)
+
+        # Perform GET
+        response = self.client.get(reverse('orders:order_history'), ECOMMERCE_REQUEST_GET_PARAMETERS)
+
+        # Check 200 OK
+        self.assertEqual(response.status_code, 200)
+
+    def test_ecommerce_view_returns_expected_response(self, mock_ecommerce_client):
+        """Check authorized user querying orders receive an expected response."""
+
+        # Login
+        self.client.login(username=self.test_user_username, password=self.test_user_password)
+
+        # Perform GET
+        response = self.client.get(reverse('orders:order_history'), ECOMMERCE_REQUEST_GET_PARAMETERS)
+
+        # Check expected response
+        self.assertEqual(response.json(), ECOMMERCE_REQUEST_EXPECTED_RESPONSE)
+
+    def test_ecommerce_view_passes_username(self, mock_ecommerce_client):
+        """Check logged in user's username is passed to the ecommerce client."""
+
+        # Login
+        self.client.login(username=self.test_user_username, password=self.test_user_password)
+
+        # Check username is coming from user, not request
+        self.assertNotEqual(ECOMMERCE_REQUEST_GET_PARAMETERS['username'], self.test_user_username)
+
```

Comment:
^ This bit (lines 104-105) confused me reading through the first time.  The comment in 104 sounds like what we're checking down in 113/114, while the assertion on 105 seems like it's just demonstrating that the locally logged-in user is different from the one in ECOMMERCE_REQUEST_GET_PARAMETERS, is that right? Or am I misunderstanding those lines?

Reply:
Yeah, you're right, this is confusing. I added it out of fear that the test could accidentally be green if `ECOMMERCE_REQUEST_GET_PARAMETERS['username']` is ever changed to be the same as `self.test_user_username` (or vice-versa).

In hindsight, this has become a tradeoff between code legibility and (far-fetched) test resiliency, so I'll remove those lines. Thanks!

Reply:
Done, see [diff](https://github.com/edx/commerce-coordinator/compare/dfb5f18df3f4b6a29084ed527186bec6fae7062a..62a256fd7e133c8f3dccfc7c0b1371d38238738d).

Reply:
Yes it's definitely a weird balance sometimes- I like your new comment that this local user is different- thanks!